subprojects/common/http-client/src/main/kotlin/com/avito/http/internal/StatsHttpEventListener.kt

```diff
@@ -47,30 +48,36 @@ internal class StatsHttpEventListener(
     }
 
     private fun serviceMetric(request: Request): SeriesName {
-        val metadata = request.tag(RequestMetadata::class.java)
+        return requestMetadataProvider.provide(request).fold(
+            onSuccess = { metadata ->
+                SeriesName.create(
+                    "service",
+                    metadata.serviceName,
+                    metadata.methodName
+                )
+            },
+            onFailure = { throwable ->
+                val urlWithoutParams = request.url.newBuilder()
+                    .query(null)
+                    .build()
 
-        return if (metadata == null) {
-            val urlWithoutParams = request.url.newBuilder()
-                .query(null)
-                .build()
+                val seriesName = SeriesName.create(
+                    "service",
```

Comment:
Did you create a view on dashboard to notice the amount of that problems?

Reply:
yes, that was the first one i created; look on top of dashboard